### ISSUE TITLE
window.Quill undefined when use ImageResize module

### ISSUE THREAD

**Issue by Insolita** [davidroyer/vue2-editor#91]
When i try to use ImageResize module, (with any variants https://davidroyer.github.io/vue2-editor/#/?id=how-to-use-custom-quill-modules  1,2) 
I get error **window.Quill undefined** 
There are no problems if use only ImageDrop module 
Vue ^2.5.13
"vue2-editor": "^2.4.3"

**Comment by bassemlhm** [comment]
See solution here https://github.com/kensnyder/quill-image-resize-module/issues/7#issuecomment-304463415

**Comment by fuping2311** [comment]
I tried every solution in  [https://github.com/kensnyder/quill-image-resize-module/issues/7#issuecomment-304463415](url)，it still unsolved

**Comment by jaskiratr** [comment]
Was anyone able to solve this yet? Facing the same issue here.

**Comment by jordandc1** [comment]
> Was anyone able to solve this yet? Facing the same issue here.

Try this one https://github.com/surmon-china/vue-quill-editor

**Comment by jaskiratr** [comment]
Thank you @jordandc1 
I'm using Nuxt SSR, so this https://github.com/surmon-china/vue-quill-editor/issues/171#issuecomment-370253411 also helped vue-quill-editor

**Comment by sushiljic** [comment]
As per the documentation,
https://www.vue2editor.com/examples/#how-to-use-custom-quill-modules
If there is any issue of quill, following code need to be implemented in webpack
since i have used vue cli 3, i have added code on my vue.config.js where at the top
const webpack = require('webpack').
`configureWebpack: {
    plugins: [
        new webpack.ProvidePlugin({
            'window.Quill': 'quill/dist/quill.js',
            'Quill': 'quill/dist/quill.js',
        }),

    ]
    }`

**Comment by samosbor** [comment]
@sushiljic Your solution worked for me. Here is my full vue.config.js file:
`const webpack = require('webpack')
module.exports = {
    configureWebpack: {
        plugins: [
            new webpack.ProvidePlugin({
                'window.Quill': 'quill/dist/quill.js',
                'Quill': 'quill/dist/quill.js',
            })]
    }
}`

I used the Import and Register Yourself way to use the custom quill modules.

**Comment by rajesh-h** [comment]
Hi Guys,

how can i achieve this on my nuxt app? how/where can i ad this config? I am struggling to use Quill module

**Comment by davidroyer** [comment]
This issue is stale because it has been open for 60 days with no activity.

**Comment by davidroyer** [comment]
This issue was closed because it has been inactive for 7 days since being marked as stale.

**Comment by RaihanShezan** [comment]
I am using `react-quill` (v2.0.0) with `quill-image-resize-module-react` (v3.0.0) on nextjs & I faced the same problem.... 

1. First I found out that I need to `dynamic import` React-Quill or its parent component because this package depends on client side object like window. So here's what I did to import my Editor component which includes react-quill - 
```js
import dynamic from 'next/dynamic'
const Editor = dynamic(import('components/Editor'), { ssr: false })
```

2. The image resizing worked well until I clicked on any image & pressed `DEL` or `BACKSPACE`, at which point I found this `window.Quill is undefined` error. All I did was add `window.Quill = Quill` to my code & it was good to go again. 
```js
import ImageResize from 'quill-image-resize-module-react'
import ReactQuill, { Quill } from 'react-quill'
import 'react-quill/dist/quill.snow.css'

window.Quill = Quill
Quill.register('modules/imageResize', ImageResize)
```

Hope this helps someone save some time debugging 😊
